Some mappings for falling blocks, food items and ParticleArgumentType

Juuxel opened the pull request:
(I have no idea why those previous commits are there...)

Juuxel committed:
Update BucketItem.mapping

Juuxel committed:
Updates to BucketItem, the Waterloggable hierarchy and block materials

Juuxel committed:
Merge remote-tracking branch 'upstream/18w50a' into 18w50a

Juuxel committed:
Merge remote-tracking branch 'upstream/18w50a' into 18w50a

Juuxel committed:
fillWithFluid -> tryFillWithFluid

Juuxel committed:
Some work on falling blocks

Juuxel committed:
Merge branch '18w50a' of https://github.com/Juuxel/yarn into 18w50a

Juuxel committed:
Delete BucketInteractable.mapping

Accidentally included by merging.

Juuxel committed:
Food items and ParticleArgumentType

SquidDev commented on mappings/net/minecraft/block/AnvilBlock.mapping:
```diff
@@ -11,8 +11,15 @@ CLASS bfz net/minecraft/block/AnvilBlock
 	METHOD <init> (Lbgs$c;)V
 		ARG 1 settings
 		ARG 2 stateBuilder
+	METHOD a configureEntity (Laox;)V
```
I wonder if something like `configureFallingBlock` would be better? `Entity` is a little vague, though I realise it's obvious once you see the method signature.

Juuxel replied:
`configureFallingBlock` could be confusing as the class itself is called FallingBlock.

shadowfacts replied:
Maybe not `configureFallingBlock`, but `configureEntity` is definitely too vague.

Juuxel replied:
Maybe `configureFallingBlockEntity`?

Juuxel replied:
I've changed it to `configureFallingBlockEntity`.

shadowfacts commented on mappings/net/minecraft/block/AnvilBlock.mapping:
```diff
@@ -11,8 +11,15 @@ CLASS bfz net/minecraft/block/AnvilBlock
+		ARG 2 pos
+	METHOD a (Lbbp;Let;Lbpm;Lbpm;)V
+		ARG 2 pos
+		ARG 3 state
```
There are two state parameters here, what's the difference? The names should reflect that

Juuxel replied:
I've accidentally mapped that state parameter, it's supposed to be unmapped for this PR.

Juuxel replied:
Fixed.

asiekierka commented:
I'm going to close this repository by next snapshot due to inactivity.

Juuxel committed:
Small FallingBlock changes

- Removed the accidentally mapped state parameter
- Renamed configureEntity -> configureFallingBlockEntity

Juuxel commented:
@shadowfacts Could you review again?

Juuxel commented:
> Create a branch for every PR :)

Yeah, I really should.
